Feat : 전세가이드 플로팅 버튼 및 6-1 단계 콘텐츠 시스템 구현

Jihye-kr opened the pull request:
## 📌 Issue

<!-- 해결하려는 이슈 번호나 주제를 명확하게 적어주세요. -->

- 관련 이슈: close #197

---

## 🛠 작업 내용

### 🎯 주요 기능
- 플로팅 액션 버튼 - 메인 페이지 하단에 전세가이드로 이동하는 버튼 추가
- 6-1 단계 콘텐츠 - 내용증명 작성 방법 가이드 구현
- 동적 z-index 관리 - 햄부기 대시보드와의 레이어링 최적화
### 🔧 기술적 개선
- TextOnly 컴포넌트 확장: 이미지, 버튼, 구조화된 콘텐츠 지원
- JSON 데이터 구조: 유연한 콘텐츠 타입 시스템
- PWA 호환성: 외부 링크 새 탭 열기 지원
- 성능 최적화: 불필요한 API 호출 제거
### 📁 변경된 파일
`app/(anon)/main/_components/floatingButtom/ `- 플로팅 버튼 컴포넌트
`app/(anon)/main/page.tsx` - 동적 z-index 관리
`app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/` - 콘텐츠 시스템
`app/(anon)/@detail/steps/[step-number]/[detail]/_components/ModalContent.tsx `- 인터페이스 확장
### 🎨 UI/UX 개선
- 메인 페이지에서 전세가이드로의 직관적인 접근
- 풍부한 콘텐츠 표시 (이미지, 버튼, 구조화된 정보)
- `hambugi` 대시보드와의 자연스러운 상호작용


## 🚀 기타 사항
<img width="300" height="977" alt="스크린샷 2025-08-22 오전 10 12 56" src="https://github.com/user-attachments/assets/b9beb074-7538-4813-b305-6f018b98551b" />
<img width="300" height="954" alt="스크린샷 2025-08-22 오전 10 13 00" src="https://github.com/user-attachments/assets/e8d348db-d962-492a-81da-22a8b3131af2" />

<!-- 리뷰어가 추가적으로 알아야 할 사항이 있다면 기재해주세요. -->


- [x] build 체크 했나요?

- [x] dev를 pull 받은 뒤 merge 했나요?

Jihye-kr committed:
fix : Hambugi 상태에 따라 메인 FloatingButton z-index 동적조절

Jihye-kr committed:
feat: TextOnly 컴포넌트 기능 확장 및 메인페이지 floating 버튼 추가

- 이미지, 버튼, 다중 서브타이틀 지원 추가
- 외부 링크 새 창 열기 기능
- 내용증명 작성 방법 상세 가이드 추가
- 메인페이지 하단에 floating 버튼 추가 (전세 가이드 시작)
- 햄부기 대시보드 열림 시 floating 버튼 z-index 조절

Jihye-kr committed:
fix: 빈 파일 정리 및 코드 정리

Jihye-kr committed:
fix: TextOnly 컴포넌트 에러 처리 로직 제거 및 JSON 데이터 직접 렌더링 구현

- useGetStepResult 훅 제거로 불필요한 API 호출 방지
- 에러 메시지 대신 JSON 데이터 직접 표시
- 불필요한 import 정리 및 컴포넌트 구조 단순화
- 6-1 단계 상세 보기에서 콘텐츠 정상 표시

icw0201 commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/TextOnly.tsx:
```diff
@@ -2,76 +2,46 @@
 
 const TextOnly = ({ data }: TextOnlyProps) => {
-  // 전역 store에서 선택된 주소 가져오기
-  const selectedAddress = useUserAddressStore((state) => state.selectedAddress);
```
이부분 왜지우신거에여?

Jihye-kr replied:
실수... 다시 복구완!💡

icw0201 commented on app/(anon)/_components/common/header/Header.tsx:
```diff
@@ -50,11 +50,14 @@ export default function Header() {
+        }`}
+        data-dashboard="true"
+      >
+        <HambugiDashboard onClose={() => setIsDashboardOpen(false)} />
```
함~부기~

Jihye-kr replied:
함부기~

icw0201 commented on app/(anon)/_components/dashboard/HambugiDashboard.styles.ts:
```diff
@@ -1,8 +1,10 @@
-  dividerTriangle: "absolute top-0 left-1/2 transform -translate-x-1/2 w-0 h-0 border-l-[12px] border-r-[12px] border-t-[12px] border-l-transparent border-r-transparent border-t-brand-white",
-  rightPanel: "w-[70%] overflow-y-auto"
+  container:
+    'h-full w-full max-w-[480px] bg-brand-white shadow-xl flex flex-col z-10000',
```
z-10000의 기개가 느껴지네요

Jihye-kr replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

icw0201 commented on app/(anon)/main/_components/floatingButtom/FloatingButton.tsx:
폴더명 오타?

Jihye-kr replied:
오타..

Jihye-kr replied:
수정완..

icw0201 reviewed (changes requested):
저도 정말 APP하고 싶네요, 없앤 것 정진 바랍니다.

Jihye-kr committed:
feat: TextOnly 컴포넌트에 사용자 진행 상황 표시 기능 복원

- useGetStepResult 훅으로 사용자 스텝 진행 상황 조회
- CircularIconBadge로 진행 상황 시각화 (match/mismatch/unchecked)
- 로딩 및 에러 상태 처리 로직 복원
- JSON 콘텐츠와 사용자 진행 상황을 모두 표시하는 하이브리드 렌더링 구현

Jihye-kr committed:
fix: 폴더명 오타 수정 (floatingButtom → floatingButton)

- app/(anon)/main/_components/floatingButtom/ → floatingButton/로 폴더명 수정
- main/page.tsx에서 import 경로 업데이트
- 폴더명 일관성 개선

icw0201 reviewed (approved):
APP

layout-SY commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/TextOnly.tsx:
```diff
@@ -6,19 +6,47 @@ import { useGetStepResult } from '@/hooks/useStepResultQueries';
 
 interface ContentSection {
   title?: string;
   subtitle?: string;
```
subtitle를 없애주세요 !

Jihye-kr replied:
수정완

layout-SY commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/TextOnly.tsx:
```diff
@@ -85,6 +113,26 @@ const TextOnly = ({ data }: TextOnlyProps) => {
+            )}
+            {section.image && (
+              <div className={styles.imageContainer}>
+                <img
```
Image next 컴포넌트 사용해주세요 !

Jihye-kr replied:
수정완

layout-SY commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/contents/TextOnly.tsx:
```diff
@@ -94,9 +142,61 @@ const TextOnly = ({ data }: TextOnlyProps) => {
+            {section.buttons && section.buttons.length > 0 && (
+              <div className={styles.buttonsContainer}>
+                {section.buttons.map((button, index) => (
+                  <button
```
Button 공용 컴포넌트 써주세요 !

Jihye-kr replied:
수정완

layout-SY commented on app/(anon)/@detail/steps/[step-number]/[detail]/_components/ModalContent.tsx:
```diff
@@ -25,35 +49,30 @@ interface StepContentData {
-  const stepNumber = pathParts[2]; // /steps/4/2 에서 4
-  const detail = pathParts[3];     // /steps/4/2 에서 2
+  const stepUrlData = parseStepUrl(pathname);
+  const stepNumber = stepUrlData?.mainNum?.toString() || '1';
```
URL 파라미터명을 step-number, detail로 변경 해주세요 !

icw0201 replied:
이거 파써에서 제가 바꿔야됫음 ㅠ ㅈㅅㅈㅅ

Jihye-kr replied:
const stepNumber = stepUrlData?.mainNum?.toString() || '1';
  const detail = stepUrlData?.subNum?.toString() || '1';
  
  이거?

Jihye-kr replied:
@icw0201 님 수정 예정

icw0201 commented:
아아앙~~~~~~ 아아아앙~~~~~~~~~~~~~~~~~~~~~ rename하기로 햇자나~~~~~~

layout-SY commented on app/(anon)/main/page.tsx:
```diff
@@ -1,21 +1,41 @@
+  // 햄부기 대시보드 열림 상태 감지
+  useEffect(() => {
+    const checkDashboardOpen = () => {
+      const dashboardElement = document.querySelector('[data-dashboard="true"]');
```
querySelector를 사용해야 하는 이유가 있었나요?

layout-SY commented on app/(anon)/main/page.tsx:
```diff
@@ -1,21 +1,41 @@
+    checkDashboardOpen();
+
+    // MutationObserver로 DOM 변화 감지
+    const observer = new MutationObserver(checkDashboardOpen);
```
이건 어떤 객체인가요?

Jihye-kr replied:
삭제완

Jihye-kr committed:
refactor: TextOnly 컴포넌트 최적화 및 z-index 구조 개선

- subtitle 속성 제거하고 subtitles 배열로 통일
- 모든 JSON 파일에서 subtitle → subtitles 구조 변경
- Next.js Image 컴포넌트 적용으로 이미지 최적화
- 모든 버튼을 공용 Button 컴포넌트로 통일
- 동적 z-index 조절 로직 제거 및 고정 z-index 구조로 단순화
- querySelector, MutationObserver, localStorage 등 DOM 직접 조작 코드 제거
- 햄부기 대시보드 z-index를 z-[60]으로 상향 조정하여 FloatingButton 위에 표시

layout-SY reviewed (approved):
APP
